pkg/datapath/linux/routing/migrate.go

```diff
@@ -0,0 +1,623 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linuxrouting
+
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/datapath/linux/linux_defaults"
+	"github.com/cilium/cilium/pkg/revert"
+
+	"github.com/vishvananda/netlink"
+)
+
+// MigrateENIDatapath migrates the egress rules inside the Linux routing policy
+// database (RPDB) for ENI IPAM mode. It will return the number of rules that
+// were successfully migrated and the number of rules we've failed to migrated.
+// A -1 is returned for the failed number of rules indicating we couldn't even
+// start the migration.
```

Comment:
nit, without conditionals it sounds like `failed` is always `-1`:

```suggestion
// A -1 may be returned for the failed number of rules indicating we couldn't even
// start the migration.
```

or

```suggestion
// A -1 is returned for the failed number of rules if we couldn't even
// start the migration.
```

Reply:
Taking the latter, thanks